### PR TITLE
Allows to choose SSL context for SMTP connection

### PR DESCRIPTION
**Pull request by potiuk**
This change add two options to choose from when SSL SMTP connection is created:

* default - for balance between compatibility and security
* none - in case compatibility with existing infrastructure is preferred

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/config_templates/config.yml [hunk 1]
```diff
@@ -1862,6 +1862,22 @@ email:
...
+        compatibility and security, it however requires that certificates in your operating system are
+        updated and that SMTP/IMAP servers of yours have valid certificates that have corresponding public
+        keys installed on your machines. You can switch it to "none" if you want to disable checking
+        of the certificates, but it is not recommended as it allows MITM (man-in-the-middle) attacks
+        if your infrastructure is not sufficiently secured. It should only be set temporarily while you
+        are fixing your certificate configuration. This can be typically done by upgrading to newer
+        version of the operating system you run Airflow components on,by upgrading/refreshing proper
+        certificates in the OS or by updating certificates for your mail servers.
+      type: string
+      version_added: 2.7.0
```
Comment (eladkal) [inline airflow/config_templates/config.yml:1878]: Mmm do we want to add this as config?
We have WIP PR to extract smtp logic from core to providers
https://github.com/apache/airflow/pull/30531/files

cc @hussein-awala

Comment (potiuk) [reply airflow/config_templates/config.yml:1878]: Not yet. This is 2.7.0 change. We will follow up with providers as next step.

**Review by pankajkoti — COMMENTED** [review]
Looks great to me, but not enough experience with SSL, certificates or SMTP, sorry cannot add insights or ensure much here :)

**Review by ephraimbuddy — APPROVED** [review]

File: airflow/config_templates/config.yml (hunk 1 above)
Comment (hussein-awala) [reply airflow/config_templates/config.yml:1878]: it's fine to merge this one

**Review by hussein-awala — APPROVED** [review]

**Pull request merged**